hera_cal/lst_stack/config.py

```diff
@@ -386,103 +256,644 @@ def make_lst_bin_config_file(
+
+        # Turn matched_files back into a list of lists of strings.
+        matched_files = [
+            [[str(m.path) for m in night] for night in outfiles]
+            for outfiles in matched_files
+        ]
+
+        antpairs, pols = get_all_antpairs(
+            data_files=self.data_files,
+            include_autos=True,
+            ignore_ants=self.ignore_ants,
+            only_last_file_per_night=self.antpairs_from_last_file_each_night,
+            redundantly_averaged=self.is_redundantly_averaged,
+            reds=self.reds,
+            blts_are_rectangular=self.datameta.blts_are_rectangular,
+            time_axis_faster_than_bls=self.datameta.time_axis_faster_than_bls,
         )
 
-    matched_files = [
-        [[str(m.path) for m in night] for night in outfiles]
-        for outfiles in matched_files
+        return LSTConfig(
+            config=self,
+            lst_grid=lst_grid,
+            matched_files=matched_files,
+            antpairs=[tuple(ap) for ap in antpairs if ap[0] != ap[1]],
+            autos=[tuple(ap) for ap in antpairs if ap[0] == ap[1]],
+            pols=pols,
+            inpaint_files=apply_filename_rules(
+                matched_files, self.where_inpainted_file_rules, missing='raise'
+            ) if self.where_inpainted_file_rules else None,
+            calfiles=apply_filename_rules(
+                matched_files, self.calfile_rules, missing='raise'
+            ) if self.calfile_rules else None,
+            properties={
+                "lst_branch_cut": lst_branch_cut,
+                "blts_are_rectangular": self.datameta.blts_are_rectangular,
+                "time_axis_faster_than_bls": self.datameta.time_axis_faster_than_bls,
+                "x_orientation": self.datameta.x_orientation,
+                "first_jd": self.get_earliest_jd_in_set(),
+            }
+        )
+
+    def write(self, group: h5py.Group):
+        dct = attrs.asdict(self)
+
+        for k, v in dct.items():
+            if k == 'data_files':
+                for night, files in enumerate(v):
+                    group.create_dataset(f"night_{night}", data=[str(f) for f in files])
+            elif k in ('calfile_rules', 'where_inpainted_file_rules'):
+                if v:
+                    group.create_dataset(k, data=v)
+            else:
+                group.attrs[k] = v
+
+    @classmethod
+    def read(cls, group: h5py.Group):
+        dct = dict(group.attrs.items())
+        n_nights = len([k for k in group.keys() if k.startswith("night_")])
+        dct["data_files"] = []
+        for night in range(n_nights):
+            dct['data_files'].append([f.decode() for f in group[f"night_{night}"][()]])
+
+        for k in ("calfile_rules", "where_inpainted_file_rules"):
+            if k in group:
+                dct[k] = [tuple(x.decode() for x in rule) for rule in group[k][()]]
+            else:
+                dct[k] = None
+
+        return cls(**dct)
```

Comment:
OK, so these exist because they are called from the `LSTConfig` object itself to write the Configurator parameters into the config file (useful for checking back on how the files were configured). They shouldn't really be called by users, so I'll make them `_` methods, and write docstrings...